src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -0,0 +1,161 @@
+#include "ray/core_worker/transport/direct_task_transport.h"
+
+namespace ray {
+
+void DoInlineObjectValue(const ObjectID &obj_id, std::shared_ptr<RayObject> value,
```

Comment:
Shouldn't this also be shared with `direct_actor_transport`?

Reply:
No, since we never inline in that case.

Reply:
Perhaps as a future optimization for inlining. I haven't thought that far ahead.

Reply:
Er...it seems like this should just be the codepath that all actor/nonactor tasks take, not just as an optimization. But we can discuss offline.